cogs/roles.py

```diff
@@ -14,17 +18,16 @@ async def giverole(self, ctx: commands.Context, existing_role: discord.Role, new
         """
         Assign a new role to all members with a specific role.
         """
-        if existing_role is None or new_role is None:
-            return await ctx.send("Please provide both the existing and new role.")
         role_members = [member for member in ctx.guild.members if existing_role in member.roles]
         for member in role_members:
             try:
                 await member.add_roles(new_role)
             except discord.Forbidden:
                 return await ctx.send("I do not have permission to add roles to members.")
             except discord.HTTPException:
-                await ctx.send(f"Could not add role to {member.name}")
-        await ctx.send("Successfully added roles.")
+                await ctx.send(f"Could not add role to {member.name}.")
+        await ctx.send(f"Successfully added roles to {len(role_members)} members.") 
```

Comment:
The issue here is that if there were errors for specific users and the role was not added, the number of members to which the role was successfully added would not be accurate.

Example: 6 members to add the Alumni role, 1 error -> Would still report successfully adding role to 6 members
Expected: Successfully added role to 5 members.